lalamo/modules/classifier.py

```diff
@@ -0,0 +1,374 @@
+    Normalization,
+    Transformer,
+    TransformerConfig,
+    TransformerForwardPassConfig,
+)
+from lalamo.modules.utils import vmap_twice
+
+from .activations import GELU, Activation, SiLU
+from .common import ForwardPassMode, LalamoModule
+from .embedding import EmbeddingBase, EmbeddingConfig
+from .linear import LinearBase, LinearConfig
+from .rope import PositionalEmbeddings
+from .transformer_layer import TransformerLayerResult
+
+__all__ = [
+    "Classifier",
+    "ClassifierActivationTrace",
+    "ClassifierConfig",
+    "ClassifierResult",
+]
+
+
+class PoolingType(StrEnum):
+    CLS = "cls"
+    MEAN = "mean"
+
+
+def activation_from_str(activation: str) -> type[Activation]:
+    supported_activations = {
+        "silu": SiLU,
+        "gelu": GELU,
+    }
+    if activation in supported_activations:
+        return supported_activations[activation]
+
+    raise ValueError(
+        f"Only activations from the following list are supported by Classifier: {supported_activations.keys()}"
+    )
+
+
+@dataclass(frozen=True)
+class PredictionHeadConfig:
+    dense_config: LinearConfig
+    activation: Activation
+    normalization_config: NormalizationConfig
+    final_linear_config: LinearConfig
+    use_dense_bias: bool
+
+    def empty(self, input_size: int, num_labels: int) -> "PredictionHead":
+        dense_layer = self.dense_config.empty(
+            input_dim=input_size,
+            output_dims=(input_size,),
+            has_biases=self.use_dense_bias,
+        )
+        norm = self.normalization_config.empty(input_size)
+        final_linear = self.final_linear_config.empty(input_dim=input_size, output_dims=(num_labels,), has_biases=True)
+
+        return PredictionHead(
+            config=self,
+            dense=dense_layer,
+            activation=self.activation,
+            norm=norm,
+            final_linear=final_linear,
+        )
+
+    def random_init(self, input_size: int, num_labels: int, key: PRNGKeyArray) -> "PredictionHead":
+        dense_key, final_linear_key = jax.random.split(key)
+        dense_layer = self.dense_config.random_init(
+            input_size, (input_size,), has_biases=self.use_dense_bias, key=dense_key
+        )
+        norm = self.normalization_config.empty(input_size)
+        final_linear = self.final_linear_config.random_init(
+            input_dim=input_size,
+            output_dims=(num_labels,),
+            has_biases=True,
+            key=final_linear_key,
+        )
+
+        return PredictionHead(
+            config=self,
+            dense=dense_layer,
+            activation=self.activation,
+            norm=norm,
+            final_linear=final_linear,
+        )
+
+
+class PredictionHead(LalamoModule[PredictionHeadConfig]):
+    dense: LinearBase
+    activation: Activation
+    norm: Normalization
+    final_linear: LinearBase
+
+    def __call__(self, inner_features: Float[Array, "batch in_channels"]) -> Float[Array, "batch n_logits"]:
```

Comment:
What is "batch" here? The input is 3d, no?

Also output shape dimension should be called logits, not n_logits here

Reply:
Also if there are no out_channels then just call in_channels channels